avro2py/codegen.py

```diff
@@ -363,10 +362,16 @@ def bridge_namespaces(from_namespace: List[str], to_namespace: List[str], target
         remaining_to_components = [c for c in paired_remaining_to_components if c is not None]
 
         if remaining_from_components:
+            level = len(remaining_from_components)
+            for namespace_name in self.namespaces.keys():
```

Comment:
I see that the access to `self.namespaces` means that `bridge_namespaces` is no longer a `@staticmethod` - since `bridge_namespaces` has some pretty complex logic that was _previously_ pure, I would ask that either:
1. We simply pass in these namespaces and revert to being a `@staticmethod`, to make it easier to reason about what this method is doing in future (e.g. we'll know it's not mutating anything), or
2. We extract the pure components of the method (i.e. https://github.com/hchasestevens/avro2py/pull/15/files#diff-dd17a87ea1ee28b0a8ef80f33739b424462dfd309d0a7eb4d58b619f0ec386f3R347-R362 ) into a new `@staticmethod` and write tests to exercise that new method specifically.

Reply:
These are passed in and it's static again.